src/components/DataTable.js

```diff
@@ -73,7 +75,23 @@ export default ajaxCaller(class DataTable extends Component {
 
     return h(Fragment, [
       !!entities && h(Fragment, [
-        childrenBefore && childrenBefore({ entities, columnSettings }),
+        div({ style: { display: 'flex' } }, [
+          childrenBefore && childrenBefore({ entities, columnSettings }),
+          div({ style: { flexGrow: 1 } }),
+          div({ style: { display: 'flex' } }, [
+            h(SearchInput, {
+              style: { width: 300, borderColor: colors.gray[3], borderRadius: '4px 0 0 4px' },
+              placeholder: 'Filter',
+              onChange: ({ target: { value } }) => this.setState({ textFilter: value }),
+              onSearch: () => this.loadData(),
```

Comment:
TOL: This is fine, though if this was structured such that the confirmed search value was written to state, we could roll it into the unified fetching logic in `componentDidMount`. That pattern is nice because it decouples the 'effect' from the state management.